src/test/java/com/ea/restaurant/service/impl/UserServiceE2ETest.java

```diff
@@ -0,0 +1,88 @@
+package com.ea.restaurant.service.impl;
+
+import com.ea.restaurant.entities.User;
+import com.ea.restaurant.repository.UserRepository;
+import com.ea.restaurant.service.UserService;
+import com.ea.restaurant.service.impl.utils.fixtures.UserFixture;
+import java.util.Arrays;
+import java.util.Optional;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+@SpringBootTest
+public class UserServiceE2ETest {
+
+  @Autowired private JdbcTemplate jdbcTemplate;
+
+  @Autowired private UserRepository userRepository;
+
+  @Autowired private UserService userService;
+
+  private User userToTest1;
+  private User userToTest2;
+
+  @BeforeEach
+  void setUp() {
+    jdbcTemplate.execute("TRUNCATE TABLE users CASCADE");
+    jdbcTemplate.execute("ALTER SEQUENCE users_id_seq RESTART");
+    userService = new UserServiceImpl(userRepository);
+    userToTest1 = UserFixture.buildUser(1L, "ea", "pg", "eapg", "1234");
+    userToTest2 = UserFixture.buildUser(2L, "ep", "gp", "ea001", "abcd1234");
+  }
+
+  @Test
+  public void ShouldCreateUser() {
+    var userSaved = this.userService.create(this.userToTest1);
+    var userReturned = this.userService.findEnabledById(userSaved.getId());
+    var expectedUser = UserFixture.buildUser(1L, "ea", "pg", "eapg", "1234");
+    Assertions.assertEquals(expectedUser, userSaved);
+  }
+
+  @Test
+  public void ShouldReturnUserById() {
+    var userSaved = this.userService.create(this.userToTest1);
+    var userReturned = this.userService.findEnabledById(userSaved.getId());
+    Assertions.assertEquals(userReturned, Optional.of(this.userToTest1));
+  }
+
+  @Test
+  public void ShouldReturnUserByUsername() {
+    this.userService.create(this.userToTest2);
+    var userReturned = this.userService.findEnabledByUsername(this.userToTest2.getUsername());
+    Assertions.assertEquals(this.userToTest2.getUsername(), userReturned.get().getUsername());
+  }
+
+  @Test
+  public void ShouldReturnAllEnabledUsers() {
+    this.userService.create(this.userToTest1);
+    this.userService.create(this.userToTest2);
+    var users = Arrays.asList(this.userToTest1, this.userToTest2);
+    var usersReturned = this.userService.findAllEnabled();
+    Assertions.assertEquals(usersReturned, users);
+  }
+
+  @Test
+  public void ShouldDeleteUser() {
+    var user1 = this.userService.create(this.userToTest1);
+    var user2 = this.userService.create(this.userToTest2);
+    var userToDelete = this.userService.findEnabledById(user1.getId());
```

Comment:
Unwrap the user if you are not going to use optional features

Reply:
I think you don't need it here, you just unwrap by doing `this.userService.findEnabledById(user1.getId()).orElseThrow(EntityNotFoundException::new)`

Reply:
Get it.